src/svgui/svgui.ts

```diff
@@ -250,6 +251,69 @@ export class Pair extends UIElement {
     }
 }
 
+/*
+ * Image for element.
+ * Backbone properties:
+ *     imageUrl: string
+ */
+export class Image extends UIElement {
+    public image: d3.Selection<any>;
+    public imageRect: d3.Selection<any>;
+
+    defaults() {
+        return _.extend(super.defaults(), {
+            imageUrl: undefined,
+            borderColor: 'green',
+            spacing: vector(10, 3),
+            padding: undefined,
+            minSize: vector(100, 100),
+        });
+    }
+    initialize() {
+        super.initialize.apply(this, arguments);
+        this.root.attr('class', 'svguiImage');
+        this.update();
+        this.imageRect = this.root.append('rect')
+            .attr('fill', '#e1e1e1')
+            .attr('stroke-width', '1')
+            .attr('stroke', this.get('borderColor'));
+        this.image = this.root.append('image')
+            .attr('x', 0)
+            .attr('y', 0)
+            .attr('width', this.get('minSize').x)
+            .attr('height', this.get('minSize').y)
+            .attr('href', this.get('imageUrl'));
+    }
+    update() {
+        if (this.image) {
```

Comment:
You could remove these `if` statements by moving `this.update()` call after setting `this.imageRect` and `this.image` fields in `initialize()`.